### ISSUE TITLE
Crash on an image sequence movie strip

### ISSUE THREAD

**Issue by jin-eld** [lukas-blecher/AutoMask#6]
Hi,

first of all - thanks for this project, I am really very excited about the possibilities that are opening up here :)

Unfortunately my first attempt did not go too well.

I imported a set of frames instead of a movie file, when I try to track the next frame blender on Linux crashes:

```
[ INFO:0] VIDEOIO: Enabled backends(6, sorted by priority): GSTREAMER(1000); V4L2(990); OPENNI(980); CV_IMAGES(970); CV_MJPEG(960); FIREWIRE(950)
[ INFO:0] OpenCV | GStreamer:../Videos/frame_001.png
[ INFO:0] OpenCV | GStreamer: mode - FILE
[ WARN:0] OpenCV | GStreamer warning: unable to query duration of stream
[ WARN:0] OpenCV | GStreamer warning: Cannot query video position: status=1, value=0, duration=-1
Writing: /tmp/automask-test.crash.txt
Segmentation fault (core dumped)
```

I assume that GStreamer expects a video and tries to parse the duration and video position, which of course does not work when dealing with an image strip.

If image strips can't be used then it'd be good to catch that and avoid the crash.

Kind regards,
Jin

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. Windows]
 - Parameters

**Additional context**

**Comment by lukas-blecher** [comment]
Hi,
thank you for your interest and the issue.
Sadly the project is far from being bug free. I think you are correct, a movie file is expected. This should be fairly easy to fix though.
I'll see what I can do tomorrow.

**Comment by lukas-blecher** [comment]
Hey, I think I've got it working now.
Uninstall and reinstall the addon. Let me know if it works

**Comment by jin-eld** [comment]
Hi! 

> Hey, I think I've got it working now.

Cool, that was fast :)



>  Uninstall and reinstall the addon. Let me know if it works

Unfortunately I ran into issues there and I am not sure how to continue; yesterday I reimported the clip as video instead of single images in order to work around this issue (#6), and then I got a bit further and failed on dependencies...

And I am not sure how to do it. I tried using blenders python and run ensurepip, then pip install all deps into the blender pyhton directory - but then blender just crashes (regardless of the add-on), I tried installing the deps on my system but then blender was not able to findn the deps, i.e. add-on excepted when loading module PIL, I tried removing blenders python dir so that blender would take python 3.7m from my system, but then blender just crashed.

Do you have any hints for me, how did you do it? 

I am on Fedora 31 btw.

You don't happen to hang around on #blender IRC on freenode, do you? :)

Kind regards,
Jin

P.S. I was trying the releases from blender.org, because blender from Fedora repos lacks ffmpeg support.

**Comment by jin-eld** [comment]
OK I fugured out a workaround - since AutoMask has a fix for frames now, I can test it with the blender version from the repo without ffmpeg. Unfortunately blender crashes when I attempt to track the next frame.

[automask-test.crash.txt](https://github.com/lukas-blecher/AutoMask/files/5378753/automask-test.crash.txt)

**Comment by lukas-blecher** [comment]
how does the `paths` variable look like in `automask.py` L7-10? do the paths point to the correct directories?

**Comment by jin-eld** [comment]
Yes... I added two paths: one to my local pip installation and one to the git checkout of the plugin:

```
paths = [
    "/home/xxx/.local/lib/python3.7/site-packages",
    "/home/xxxAutoMask"
]
```

**Comment by lukas-blecher** [comment]
> I tried using blenders python and run ensurepip, then pip install all deps into the blender pyhton directory - but then blender just crashes (regardless of the add-on)

That's weird. I don't know what could cause that. I know it is possible to run it with the blender python but I never tried it myself.
> I tried installing the deps on my system but then blender was not able to findn the deps, i.e. add-on excepted when loading module PIL

Sometimes blender needs a restart after reinstalling the addon. I assume you changed the python path (which does look perfectly fine) and then uninstalled and reinstalled the addon. But according to the crash log the correct python path is passed to blender.
> I tried removing blenders python dir so that blender would take python 3.7m from my system, but then blender just crashed.

That's not a good idea. That python has some extra stuff. Is the crash log from this try? If so try again with the blender python. Maybe before the python path was not updated

**Comment by jin-eld** [comment]
Could you hint me how you installed the dependencies for the plugin on your system? Did you pip install them into the blender python environment or how did you do it?

I don't think that your add-on is causing the crash directly in this case, its probably something dependency/python related, I guess I will compile blender myself so I can look at the full backtrace with debug symbols. But it would still help if you could provide the info how you dealt with the python deps so that I could try it your way first. Thank you!

**Comment by lukas-blecher** [comment]
I've installed the dependencies on my standalone python 3.7 and used them for other projects before starting this project. All I had to do for blender to recognize the modules was giving it the path to the site packages.

Can you verify if the dependencies are installed correctly?

**Comment by lukas-blecher** [comment]
I'm closing this issue because the addon can handle image sequences now. If you still have installation problems don't hesitate to open another issue
